packages/babel-helper-modules/src/normalize-and-load-metadata.js

```diff
@@ -0,0 +1,378 @@
+export function hasExports(metadata: ModuleMetadata) {
+  const { local, source } = metadata;
+
+  return (
+    local.size > 0 ||
+    Array.from(source).some(([, meta]) => {
+      return (
+        meta.reexports.size > 0 ||
+        meta.reexportNamespace.size > 0 ||
+        !!meta.reexportAll
+      );
+    })
+  );
+}
+
+/**
+ * Check if a given source is an anonymous import, e.g. "import 'foo';"
+ */
+export function isAnonymousImport(source: SourceModuleMetadata) {
+  return (
+    source.imports.size === 0 &&
+    source.importsNamespace.size === 0 &&
+    source.reexports.size === 0 &&
+    source.reexportNamespace.size === 0 &&
+    !source.reexportAll
+  );
+}
+
+/**
+ * Remove all imports and exports from the file, and return all metadata
+ * needed to reconstruct the module's behavior.
+ */
+export default function normalizeModuleAndLoadMetadata(
+  programPath: NodePath,
+  exportName?: string,
+  { strict = false, noInterop = false } = {},
+): ModuleMetadata {
+  if (!exportName) {
+    exportName = programPath.scope.generateUidIdentifier("exports").name;
+  }
+
+  const localData = normalizeLocalExports(programPath);
+
+  const sourceData = new Map();
+  const getData = sourceNode => {
+    const source = sourceNode.value;
+
+    let data = sourceData.get(source);
+    if (!data) {
+      data = {
+        // The "StringLiteral" AST node from the import/export statement.
+        sourceNode,
+        statements: [],
+
+        name: programPath.scope.generateUidIdentifier(
+          basename(source, extname(source)),
+        ).name,
+
+        interop: "none",
+
+        // Data about the requested sources and names.
+        imports: new Map(),
+        importsNamespace: new Set(),
+
+        // Metadata about data that is passed directly from source to export.
+        reexports: new Map(),
+        reexportNamespace: new Set(),
+        reexportAll: null,
+      };
+      sourceData.set(source, data);
+    }
+    return data;
+  };
+
+  programPath.get("body").forEach(child => {
+    if (child.isImportDeclaration()) {
+      const data = getData(child.node.source);
+      data.statements.push(child.node);
+
+      child.get("specifiers").forEach(spec => {
+        if (spec.isImportDefaultSpecifier()) {
+          if (data.interop === "none") data.interop = "default";
+
+          const localName = spec.get("local").node.name;
+
+          data.imports.set(localName, "default");
+
+          const reexport = localData.get(localName);
+          if (reexport) {
+            localData.delete(localName);
+
+            reexport.names.forEach(name => {
+              data.reexports.set(name, "default");
+            });
+          }
+        } else if (spec.isImportNamespaceSpecifier()) {
+          const localName = spec.get("local").node.name;
+
+          if (!strict) data.interop = "namespace";
+
+          data.importsNamespace.add(localName);
+          const reexport = localData.get(localName);
+          if (reexport) {
```

Comment:
You're right that it isn't necessary to consider namespace exports "reexports", but I did it because it simplified things and I couldn't think of anything that would break from this. Thoughts?

Reply:
It makes more sense to consider them reexports.